lego/apps/users/serializers/users.py

```diff
@@ -173,6 +208,29 @@ def get_valid_penalties(self, user):
         serializer = PenaltySerializer(instance=qs, many=True)
         return serializer.data
 
+    def get_photo_consents(self, user):
+        pc = PhotoConsent.objects.filter(user=user)
+
+        from datetime import datetime
+
+        today = datetime.today()
+        current_semester = ("H" if today.month > 7 else "V") + str(today.year)[2:4]
+        if pc.filter(semester=current_semester).count() < 2:
+            PhotoConsent.objects.create(
+                user=user,
+                semester=current_semester,
+                domain="SOCIAL_MEDIA",
+                is_consenting=None,
+            )
+            PhotoConsent.objects.create(
+                user=user,
+                semester=current_semester,
+                domain="WEBSITE",
+                is_consenting=None,
+            )
```

Comment:
I think it makes more sense to put this logic somewhere else, f.ex on the model or user? Especially since this code is duplicated. And the .count() is a little ambiguous. Maybe check if each specific photo_consent exists?

Reply:
Moved this logic to the PhotoConsent Model. Changed count to instead filter out the individual photo consents, and create new ones if needed. :+1: